### PR TITLE
catch bady placed root and use alternative re-rooting method.

### PR DESCRIPTION
**Pull request by geronimp**
not for merging - Working on one of Caitlins bugs

### CODE REVIEW

**Comment by wwood** [comment]
OK, thanks. No need to send a pull request unless you are after a review
though.

**Comment by geronimp** [comment]
Yes I'm after a review sorry, I wrote the message before my email

**Comment by geronimp** [comment]
I still want to merge this because I made some minor improvements to the re-rooting script. Let me know what you think

**Comment by wwood** [comment]
OK, sure. Probly should cherry-pick out the improvements only and include on a separate branch then tho.

**Comment by geronimp** [comment]
All of the changes are good, except for the mid point rerooting which is now useless. I'll just push another commit with that part removed if you dont mind.

**Comment by wwood** [comment]
A test or two might be in order.

**Comment by geronimp** [comment]
True

**Comment by geronimp** [comment]
Hey, I've added a test that (VERY) simply tests that the re-rooting is doing what it should do. Do you think any other test would help out here?

- J

File: graftm/rerooter.py [hunk 1]
```diff
@@ -86,12 +95,21 @@ def reroot(self, tree):
...
+                            length=0)
             return tree
-        else:
+        
+        elif len(children) == 2:
             raise BinaryTreeError("Tree is already binary. Something has \
-mislead me here!")
-
+-mislead me here!")
+        elif len(children) < 2:
```
Comment (wwood) [inline graftm/rerooter.py]: just else is better here I think

**Comment by wwood** [comment]
I think that's it,

**Comment by wwood** [comment]
ok go ahread and merge then, I take it the tests pass

**Comment by geronimp** [comment]
had that prerequisite error - I'll remove it in this branch as well and confirm that tests pass

**Pull request merged**